main.cpp

```diff
@@ -326,7 +244,14 @@ struct IntType
     IntType& multiply(int rhs);
     IntType& divide(int rhs);
 
+    IntType& pow(int rhs);
+    IntType& pow(const FloatType& rhs);
+    IntType& pow(const DoubleType& rhs);
+    IntType& pow(const IntType& rhs);
+
 private:
+    IntType& powInternal(int* base, int exp);
```

Comment:
The first argument is not needed.
```
 2) your powInternal() function should do something like this in its body:    *val = std::pow( *val, arg );
```
the internal `val/value` pointer is always used inside this function, and does not need to be passed as a function argument
- [ ] revise